### PR TITLE
Win10 upgrade - Added pending reboot check and errored installation checks

### PR DESCRIPTION
**Pull request by sethbrasile**
Win10 upgrade - Added pending reboot check and errored installation checks from installation script to download script. There were instance where installation is just pending reboot and ISO would end up cleaned up, and download would occur again when it is not needed

### CODE REVIEW

**Comment by sethbrasile** [comment]
Have one more change to make here, not quite ready for review. Want this check to move to right before triggering download so that various cleanup steps can't be missed.

**Comment by sethbrasile** [comment]
Ok, this is ready for review. I still need to test it, but the changes are super simple so I think review is ok any time

File: Windows_10_Staging/Powershell/Windows_10_Upgrade_V4/Windows_10_Upgrade_V4_Download.ps1 [hunk 1]
```diff
@@ -383,6 +385,27 @@ If ($jobIdExists -and !(Test-Path -Path $isoFilePath)) {
...
 }
 
+<#
+########################################
+3 last checks before triggering download
+########################################
+#>
+
+# We don't want this to repeatedly download if the machine is having an issue, so check for existence of setup errors
+If (Test-RegistryValue -Name $winSetupErrorKey) {
```
Comment (wmmatt) [inline Windows_10_Staging/Powershell/Windows_10_Upgrade_V4/Windows_10_Upgrade_V4_Download.ps1:395]: You set the value at line 65 to `WindowsSetupError` but I'm having a hard time tracking down how this function knows what reg path to go look for that value in? I'm probably just missing it but want to double check.

File: Windows_10_Staging/Powershell/Windows_10_Upgrade_V4/Windows_10_Upgrade_V4_Download.ps1 [hunk 2]
```diff
@@ -383,6 +385,27 @@ If ($jobIdExists -and !(Test-Path -Path $isoFilePath)) {
...
+    $setupErr = Get-RegistryValue -Name $winSetupErrorKey
+    $outputLog = "!Error: Windows setup experienced an error upon installation. This should be manually assessed and you should clear the value at $regPath\$winSetupErrorKey in order to make the script try again. The error output is $setupErr", $outputLog
+    Invoke-Output $outputLog
+    Return
+}
+
+# Check that this upgrade hasn't already occurred. No need to download iso if installation has already occurred
+If ((Test-RegistryValue -Name $pendingRebootForThisUpgradeKey) -and ((Get-RegistryValue -Name $pendingRebootForThisUpgradeKey) -eq 1)) {
```
Comment (wmmatt) [inline Windows_10_Staging/Powershell/Windows_10_Upgrade_V4/Windows_10_Upgrade_V4_Download.ps1:403]: Same question on reg path

**Review by wmmatt — APPROVED** [review]
Looks good!

**Pull request merged**